### ISSUE TITLE
[HOLD for payment 2023-02-08] [$1000] password field not auto focused while opening password protected PDF

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14265]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Actions Performed:

1. send the password protected PDF
2. open the PDF and click on enter password

## Expected Result:
password field should be auto focused

## Actual Result:
password field not auto focused

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] iOS / native

**Version Number:** 1.2.53-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/212223972-3ee2230e-c1af-4fed-988e-a613ab129be5.mp4

https://user-images.githubusercontent.com/43996225/212224077-88d7d5c2-c72e-4eeb-8f3d-b8cd30c7cd6e.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @gadhiyamanan 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1673438445881989

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~012d8f09607aa4fe53</li>
        <li>Upwork Job ID: 1616528888845672448</li>
        <li>Last Price Increase: 2023-01-20</li>
    </ul>
</details>

**Comment by Christinadobrzyn** [comment]
I think this is related - https://github.com/Expensify/App/issues/13929

**Comment by miljakljajic** [comment]
I'm not sure they're related, given that this one is mobile and browser resizing pertains to the web! @Christinadobrzyn

**Comment by miljakljajic** [comment]
Reproducible for me:
https://user-images.githubusercontent.com/28715117/212957960-3646d7b5-3287-41cd-aa3a-ac804c5c948f.mp4

Adding engineering to determine if we're good for this to be worked on externally.

**Comment by deetergp** [comment]
This can be done Externally.

**Comment by jayeshmangwani** [comment]
### Proposal

https://github.com/Expensify/App/blob/aff83ef1895d48603f9610bf56bd06722b8e10c2/src/components/PDFView/index.native.js#L169-L174


```diff

 <PDFPasswordForm 

     onSubmit={this.attemptPDFLoadWithPassword}

     onPasswordUpdated={() => this.setState({isPasswordInvalid: false})} 

     isPasswordInvalid={this.state.isPasswordInvalid}     shouldShowLoadingIndicator={this.state.shouldShowLoadingIndicator} 

+.   shouldAutofocusPasswordField

```

We can add `shouldAutofocusPasswordField` prop here as we are setting autoFocus based on this param in PDFPasswordForm,
so we can add shouldAutofocusPasswordField here and it will be autoFocus at PDFPasswordForm

https://github.com/Expensify/App/blob/aff83ef1895d48603f9610bf56bd06722b8e10c2/src/components/PDFView/PDFPasswordForm.js#L123

**Comment by gadhiyamanan** [comment]
*Proposal*

remove `shouldAutofocusPasswordField`from below file  and pass `autoFocus`  as `true`
https://github.com/Expensify/App/blob/0f744b3435f27e86bf4c70590307e3b2ff1a234d/src/components/PDFView/PDFPasswordForm.js#L18-L19
    
https://github.com/Expensify/App/blob/0f744b3435f27e86bf4c70590307e3b2ff1a234d/src/components/PDFView/PDFPasswordForm.js#L123

remove `shouldAutofocusPasswordField` prop from below file 

https://github.com/Expensify/App/blob/0f744b3435f27e86bf4c70590307e3b2ff1a234d/src/components/PDFView/index.js#L152-L158

**Comment by gadhiyamanan** [comment]
**Proposal**

remove `shouldAutofocusPasswordField`from below file  and pass `autoFocus`  as `true`
https://github.com/Expensify/App/blob/0f744b3435f27e86bf4c70590307e3b2ff1a234d/src/components/PDFView/PDFPasswordForm.js#L18-L19
    
https://github.com/Expensify/App/blob/0f744b3435f27e86bf4c70590307e3b2ff1a234d/src/components/PDFView/PDFPasswordForm.js#L123

remove `shouldAutofocusPasswordField` prop from below file 

https://github.com/Expensify/App/blob/0f744b3435f27e86bf4c70590307e3b2ff1a234d/src/components/PDFView/index.js#L152-L158

**Comment by s77rt** [comment]
I think I agree with @daraksha-dk to remove the prop `shouldAutofocusPasswordField` entirely and always focus the input. This will fix the issue on native and mWeb (not mentioned on the issue but it's also affected). However before doing so we should investigate why it was added in the first place, and it looks as it has been initially introduced by https://github.com/Expensify/App/pull/9141 in this commit https://github.com/Expensify/App/commit/ef124b8265b0c2e8c8fd7d9d9b3dcafb90949876 to fix https://github.com/Expensify/App/issues/8595 but it's not clear what part this commit is supposed to fix.

**Comment by sobitneupane** [comment]
I am not sure if it is really a bug. There are components like Composer which autofocuses on web and desktop but not on native and mWeb.

As mentioned by @s77rt, it was intentionally added in https://github.com/Expensify/App/commit/ef124b8265b0c2e8c8fd7d9d9b3dcafb90949876.

So, before proceeding I want to get confirmation from @MonilBhavsar that if it is something we want to fix.

**Comment by MonilBhavsar** [comment]
@sobitneupane good point.
I am also not sure and would like to clarify on that before we move forward. Have asked authors and reviewers on the linked PR https://github.com/Expensify/App/pull/9141/files#r1083803722

**Comment by sobitneupane** [comment]
So, looks like the reason to disable autofocus was not to let two buttons (Confirm and Send) appear over the keyboard while sending an attachment.(https://github.com/Expensify/App/pull/9141#issuecomment-1143998849)

>I tend to think we'd want to remove the autofocus, mostly because on mobile, the keyboard layout with the two buttons feels a bit strange.
>
>Is it possible to remove auto-layout but have the keyboard launch on top of the bottom green send button when the password field is focused?

Currently, when the password field is focused and Keyboard appears, the Send button disappears on small screen. So, autofocus should do no harm.
 
 We can go with @daraksha-dk's [proposal](https://github.com/Expensify/App/issues/14265#issuecomment-1398879072) if we decide to autofocus.
 
 🎀👀🎀 C+ reviewed
 
 cc: @MonilBhavsar

**Comment by MonilBhavsar** [comment]
> Currently, when the password field is focused and Keyboard appears, the Send button disappears on small screen. So, autofocus should do no harm

@daraksha-dk do you mind sharing a screenshot showcasing this ^

**Comment by MonilBhavsar** [comment]
Thanks! Look good to me.
@sobitneupane I like the proposal and I agree with removing that prop. What do you think?

**Comment by s77rt** [comment]
@MonilBhavsar Let's not remove the prop, otherwise the field will always get focused even if we add another field in the future that should be focused instead.
It won't matter much for now, but as a future proof solution maybe just pass `shouldAutofocusPasswordField` without any conditions.

**Comment by sobitneupane** [comment]
@MonilBhavsar Yes, I also like the idea of removing the prop. We can add it later if required.

**Comment by MonilBhavsar** [comment]
I agree, we can add later whenever required

**Comment by miljakljajic** [comment]
@daraksha-dk can I have your Upwork profile so I can link you to the job for payment? @mananjadhav I sent you a contract for the reporting bonus.

**Comment by mananjadhav** [comment]
@miljakljajic I am not the C+ here. I think you meant to tag @sobitneupane

**Comment by MonilBhavsar** [comment]
sorry, merged now

**Comment by sobitneupane** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [x]  The PR that introduced the bug has been identified. Link to the PR:

https://github.com/Expensify/App/pull/9141

> * [x] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

https://github.com/Expensify/App/pull/9141#discussion_r1083803722

> * [x] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

The change was intentionally made in https://github.com/Expensify/App/pull/9141 which is no longer needed.

**Comment by miljakljajic** [comment]
Reviewing payment steps now.

**Comment by miljakljajic** [comment]
@sobitneupane please accept my offer!

**Comment by miljakljajic** [comment]
proposed the regression check here: https://expensify.slack.com/archives/C01SKUP7QR0/p1676311527561559

**Comment by miljakljajic** [comment]
Regression check added, contributors paid - closing out! https://expensify.testrail.io/index.php?/cases/view/1971194
